File: LC/lc121/lc121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let cur = prices[0], prof = 0;
  for (i = 0; i < prices.length; i++){
      if (prices[i] < cur){
          cur = prices[i];
      } else if (prices[i] - cur > prof){
          prof = prices[i] - cur;
      }
  }
  return prof;
};

File: LC/lc121/lc121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        arr = list()
        for i in range(1,len(prices)):
            arr.append(prices[i] - prices[i-1])
        maxSoFar = 0
        maxCont = 0
        for i in range(len(arr)):
            maxCont = max(maxCont + arr[i], arr[i])
            maxSoFar = max(maxSoFar, maxCont)
        return maxSoFar
        